lorawan/src/subnet.rs

```diff
@@ -47,16 +47,14 @@ fn id_len(netclass: u8) -> u32 {
         .unwrap_or(&0)
 }
 
-fn subnet_addr_to_netid(subnetaddr: u32, netid_list: Vec<u32>) -> u32 {
-    for item in netid_list.clone() {
-        if subnet_addr_within_range(subnetaddr, item, netid_list.clone()) {
-            return item;
-        }
-    }
-    0
+fn subnet_addr_to_netid(subnetaddr: u32, netid_list: &[u32]) -> u32 {
+    *netid_list
+        .iter()
+        .find(|item| subnet_addr_within_range(subnetaddr, **item, netid_list))
+        .unwrap_or(&0)
```

Comment:
Is 0 the correct default case here, or should it be None?

Reply:
0 is what the original code did. I’d return an Option but that may not match the code in the spec? @mikev ?

Reply:
Rust has an Option?  Nice, I did not know that.  I'll revisit the return type for this function.

Reply:
Sure does. I'd also actually make types for netid/devaddr instead of just returning u32 for both to make it clearer what you're talking about. You can then also tie functions to those types which may make this all clearer to understand.